src/services/rfqm_service.ts

```diff
@@ -648,39 +648,52 @@ export class RfqmService {
         }
         const { calldata, makerUri, order, fee } = job;
 
-        // Start processing
-        await this._dbUtils.updateRfqmJobAsync(orderHash, {
-            status: RfqmJobStatus.PendingProcessing,
-        });
-
-        // Validate w/Eth Call
-        try {
-            await this._blockchainUtils.decodeMetaTransactionCallDataAndValidateAsync(calldata!, workerAddress);
-        } catch (e) {
-            RFQM_JOB_FAILED_ETHCALL_VALIDATION.inc();
-            logger.warn({ error: e, orderHash }, 'The eth_call validation failed');
-            // Terminate with an error transition
+        // update status to processing if it's a fresh job
+        if (job.status === RfqmJobStatus.PendingEnqueued) {
             await this._dbUtils.updateRfqmJobAsync(orderHash, {
-                status: RfqmJobStatus.FailedEthCallFailed,
+                status: RfqmJobStatus.PendingProcessing,
             });
-            return;
         }
 
-        // Get last look from MM
-        const takerTokenFillAmount = this._blockchainUtils.getTakerTokenFillAmountFromMetaTxCallData(calldata!);
-        const submitRequest: SubmitRequest = {
-            order: storedOrderToRfqmOrder(order!),
-            orderHash,
-            fee: storedFeeToFee(fee!),
-            takerTokenFillAmount,
-        };
-
-        const shouldProceed = await this._quoteServerClient.confirmLastLookAsync(makerUri!, submitRequest);
-        logger.info({ makerUri, shouldProceed, orderHash }, 'Got last look response from market maker');
+        if (job.lastLookResult === null) {
```

Comment:
We can remove nesting by framing this as a guard clause. I.e., check the illegal case first:

```
if (job.lastLookResult === false) {
  await this._dbUtils.updateRfqmJobAsync(orderHash, {
    status: RfqmJobStatus.FailedLastLookDeclined,
  });
}

// Validate w/Eth Call
// verify the order is fillable before confirming with MM
try {
...
```

Reply:
It's a technique favored by the Go community: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

Reply:
it's checking here if the lastLook has been completed or not, if not it should eth_call and send the last look request, if not skip, so not sure how to avoid nesting on the skipping vs not

Reply:
oooh, this is my bad - I misunderstood the intent - we want to avoid calling the MM twice for the same last look, so the logic goes like:
- if `lastLookResult === null`, perform the last look
- if `lastLookResult === true`, skip ahead
- if `lastLookResult === false`, terminate

This makes sense.

Wondering if we can use a log statement to help us document the "skip" 

```
} else {
  logger.info('last look previously accepted, skipping ahead to submission');
}
```